modules/http/src/test/java/org/mule/module/http/internal/request/grizzly/ProxyParamsTestCase.java

```diff
@@ -0,0 +1,160 @@
+import org.glassfish.grizzly.http.server.Request;
+import org.glassfish.grizzly.http.server.Response;
+import org.junit.*;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.mule.api.MuleException;
+import org.mule.module.http.internal.ParameterMap;
+import org.mule.module.http.internal.domain.request.DefaultHttpRequest;
+import org.mule.module.http.internal.domain.request.HttpRequest;
+import org.mule.module.http.internal.request.DefaultProxyConfig;
+import org.mule.module.http.internal.request.HttpClientConfiguration;
+import org.mule.tck.junit4.AbstractMuleTestCase;
+import org.mule.tck.junit4.rule.DynamicPort;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Arrays;
+import java.util.Collection;
+
+import static org.eclipse.jetty.http.HttpMethod.GET;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.fail;
+
+@RunWith(Parameterized.class)
+public class ProxyParamsTestCase extends AbstractMuleTestCase
+{
+
+    public DynamicPort dynamicPort = new DynamicPort("port");
+    public DynamicPort proxyPort = new DynamicPort("proxyPort");
+
+    private static final String EXPECTED_RESULT = "result";
+    private static final String PROTOCOL = "http";
+    private static final String HOST = "localhost";
+    private static final String PATH = "retrieve";
+    private static final int TIMEOUT = -1;
+    private static final Boolean FOLLOW_REDIRECTS = Boolean.TRUE;
+    private MockServer server = new MockServer(dynamicPort.getNumber());
+    private GrizzlyHttpClient httpClient;
+
+    @Parameterized.Parameter()
+    public String nonProxyHosts;
+
+    @Parameterized.Parameters(name = "{0}")
+    public static Collection<Object[]> parameters()
+    {
+        return Arrays.asList(new Object[][] {
+                {""},
+                {"localhost"},
+                {"other|localhost"}
+        });
+    }
+    
+    @After
+    public void after() throws Exception
+    {
+        server.stop();
+        httpClient.stop();
+    }
+
+    @Test
+    public void testProxyNonProxyParams() throws Exception {
+        createProxyAndClient(nonProxyHosts);
+
+        final HttpRequest request = new DefaultHttpRequest(
+                PROTOCOL + "://" + HOST + ":" + dynamicPort.getNumber() + "/" + PATH, null, GET.asString(),
+                new ParameterMap(), new ParameterMap(), null);
+
+        if (nonProxyHosts != null && !nonProxyHosts.isEmpty()) {
+            // as we set the nonProxyHosts to localhost, the proxy will not block this request
+            InputStream responseStream = httpClient.sendAndReceiveInputStream(request, TIMEOUT, FOLLOW_REDIRECTS, null);
+            assertNotNull(responseStream);
+            assertEquals(EXPECTED_RESULT, IOUtils.toString(responseStream));
+        } else {
+            try {
+                httpClient.sendAndReceiveInputStream(request, TIMEOUT, FOLLOW_REDIRECTS, null);
```

Comment:
Instead of asserting that the request fails you should set up the proxy and check that the request went through it. `AbstractHttpRequestTestCase` has a server already so you won't need the mock one here but you will need the proxy one.